### PR TITLE
Update key value processor documentation with new config options

### PR DESCRIPTION
**Pull request by kkondaka**
### Description
Update key value processor documentation with the following new config options.
`strict_grouping`
`string_literal_character`
`key_value_when`

### Version
_all._

### Checklist
- [ X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _data-prepper/pipelines/configuration/processors/key-value.md [hunk 1]
```diff
@@ -35,6 +35,9 @@ You can use the `key_value` processor to parse the specified field into key-valu
...
 | value_grouping | Specifies whether to group values using predefined value grouping delimiters: `{...}`, `[...]', `<...>`, `(...)`, `"..."`, `'...'`, `http://... (space)`, and `https:// (space)`. If this flag is enabled, then the content between the delimiters is considered to be one entity and is not parsed for key-value pairs. Default is `false`. If `value_grouping` is `true`, then `{"key1=[a=b,c=d]&key2=value2"}` parses to `{"key1": "[a=b,c=d]", "key2": "value2"}`. |
 | drop_keys_with_no_value | Specifies whether keys should be dropped if they have a null value. Default is `false`. If `drop_keys_with_no_value` is set to `true`, then `{"key1=value1&key2"}` parses to `{"key1": "value1"}`. |
+| strict_grouping | Specifies whether strict grouping should be enabled when `value_grouping` or `string_literal_character` options are used. Default is `false`. | If enabled, groups with unmatched end characters results in error. The event is ignored after error is logged.|
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/key-value.md:38]:
```suggestion
| strict_grouping | Specifies whether strict grouping should be enabled when `value_grouping` or `string_literal_character` options are used. Default is `false`. | If enabled, groups with unmatched end characters results in errors. The event is ignored after error is logged. |
```

File: _data-prepper/pipelines/configuration/processors/key-value.md [hunk 2]
```diff
@@ -35,6 +35,9 @@ You can use the `key_value` processor to parse the specified field into key-valu
...
 | value_grouping | Specifies whether to group values using predefined value grouping delimiters: `{...}`, `[...]', `<...>`, `(...)`, `"..."`, `'...'`, `http://... (space)`, and `https:// (space)`. If this flag is enabled, then the content between the delimiters is considered to be one entity and is not parsed for key-value pairs. Default is `false`. If `value_grouping` is `true`, then `{"key1=[a=b,c=d]&key2=value2"}` parses to `{"key1": "[a=b,c=d]", "key2": "value2"}`. |
 | drop_keys_with_no_value | Specifies whether keys should be dropped if they have a null value. Default is `false`. If `drop_keys_with_no_value` is set to `true`, then `{"key1=value1&key2"}` parses to `{"key1": "value1"}`. |
+| strict_grouping | Specifies whether strict grouping should be enabled when `value_grouping` or `string_literal_character` options are used. Default is `false`. | If enabled, groups with unmatched end characters results in errors. The event is ignored after error is logged. |
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/key-value.md:38]:
```suggestion
| strict_grouping | Specifies whether strict grouping should be enabled when `value_grouping` or `string_literal_character` options are used. Default is `false`. | If enabled, groups with unmatched end characters results in errors. The event is ignored after errors are logged. |
```

File: _data-prepper/pipelines/configuration/processors/key-value.md [hunk 3]
```diff
@@ -35,6 +35,9 @@ You can use the `key_value` processor to parse the specified field into key-valu
...
 | drop_keys_with_no_value | Specifies whether keys should be dropped if they have a null value. Default is `false`. If `drop_keys_with_no_value` is set to `true`, then `{"key1=value1&key2"}` parses to `{"key1": "value1"}`. |
+| strict_grouping | Specifies whether strict grouping should be enabled when `value_grouping` or `string_literal_character` options are used. Default is `false`. | If enabled, groups with unmatched end characters results in errors. The event is ignored after errors are logged. |
+| string_literal_character | A single quote(`'`) or double quote(`"`) can be configured as string literal character. Default is Null. | If this option is used, the text between the single/double quote is ignored and is not used for key value parsing. An input like `text1 "key1=value1" text2 key2=value2` parses to `{"key2": "value2"}`.|
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/key-value.md:39]:
```suggestion
| string_literal_character | Can be set to either a single quotation mark (`'`) or a double quotation mark (`"`). Default is `null`. | When this option is used, any text contained within the specified quotation mark character will be ignored and excluded from key-value parsing. For example, `text1 "key1=value1" text2 key2=value2` would parse to `{"key2": "value2"}`. | 
```

File: _data-prepper/pipelines/configuration/processors/key-value.md [hunk 4]
```diff
@@ -35,6 +35,9 @@ You can use the `key_value` processor to parse the specified field into key-valu
...
 | drop_keys_with_no_value | Specifies whether keys should be dropped if they have a null value. Default is `false`. If `drop_keys_with_no_value` is set to `true`, then `{"key1=value1&key2"}` parses to `{"key1": "value1"}`. |
+| strict_grouping | Specifies whether strict grouping should be enabled when `value_grouping` or `string_literal_character` options are used. Default is `false`. | If enabled, groups with unmatched end characters results in errors. The event is ignored after errors are logged. |
+| string_literal_character | Can be set to either a single quotation mark (`'`) or a double quotation mark (`"`). Default is `null`. | When this option is used, any text contained within the specified quotation mark character will be ignored and excluded from key-value parsing. For example, `text1 "key1=value1" text2 key2=value2` would parse to `{"key2": "value2"}`. | 
+| key_value_when | A [conditional expression](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/), such as `/some-key == "test"'`. that will be evaluated to determine whether the processor will be run on the event. |
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/key-value.md:40]:
```suggestion
| key_value_when | Allows you to specify a [conditional expression](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/), such as `/some-key == "test"`, which will be evaluated to determine if the processor should be applied to the event or not. |
```

**Review by vagimeli — APPROVED** [review]
Doc review complete with some rewrites for clarity. Moving to editorial. I'll address any editorial comments and then merge the PR. Thank you.

File: _data-prepper/pipelines/configuration/processors/key-value.md [hunk 5]
```diff
@@ -35,6 +35,9 @@ You can use the `key_value` processor to parse the specified field into key-valu
...
 | value_grouping | Specifies whether to group values using predefined value grouping delimiters: `{...}`, `[...]', `<...>`, `(...)`, `"..."`, `'...'`, `http://... (space)`, and `https:// (space)`. If this flag is enabled, then the content between the delimiters is considered to be one entity and is not parsed for key-value pairs. Default is `false`. If `value_grouping` is `true`, then `{"key1=[a=b,c=d]&key2=value2"}` parses to `{"key1": "[a=b,c=d]", "key2": "value2"}`. |
 | drop_keys_with_no_value | Specifies whether keys should be dropped if they have a null value. Default is `false`. If `drop_keys_with_no_value` is set to `true`, then `{"key1=value1&key2"}` parses to `{"key1": "value1"}`. |
+| strict_grouping | Specifies whether strict grouping should be enabled when `value_grouping` or `string_literal_character` options are used. Default is `false`. | If enabled, groups with unmatched end characters results in errors. The event is ignored after errors are logged. |
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/key-value.md:38]:
```suggestion
| strict_grouping | Specifies whether strict grouping should be enabled when the `value_grouping` or `string_literal_character` options are used. Default is `false`. | When enabled, groups with unmatched end characters yield errors. The event is ignored after the errors are logged. |
```

File: _data-prepper/pipelines/configuration/processors/key-value.md [hunk 6]
```diff
@@ -35,6 +35,9 @@ You can use the `key_value` processor to parse the specified field into key-valu
...
+| string_literal_character | Can be set to either a single quotation mark (`'`) or a double quotation mark (`"`). Default is `null`. | When this option is used, any text contained within the specified quotation mark character will be ignored and excluded from key-value parsing. For example, `text1 "key1=value1" text2 key2=value2` would parse to `{"key2": "value2"}`. | 
+| key_value_when | Allows you to specify a [conditional expression](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/), such as `/some-key == "test"`, which will be evaluated to determine if the processor should be applied to the event. |
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/key-value.md:40]:
```suggestion
| key_value_when | Allows you to specify a [conditional expression](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/), such as `/some-key == "test"`, that will be evaluated to determine whether the processor should be applied to the event. |
```

**Review by natebower — COMMENTED** [review]
@vagimeli @kkondaka Please see my changes and let me know if you have any questions. Thanks!

**Pull request merged**